tests/ops/test_channel_ops.py

```diff
@@ -41,26 +41,37 @@
 class TestChannels:
     """Tests for the quantum channels"""
 
-    @pytest.mark.parametrize("ops", ch_list)
-    @pytest.mark.parametrize("p", [0, 0.1, 1])
+    @pytest.mark.parametrize(
+        "interface",
+        [
+            None,
+            pytest.param("autograd", marks=pytest.mark.autograd),
+            pytest.param("tensorflow", marks=pytest.mark.tf),
+            pytest.param("jax", marks=pytest.mark.jax),
+            pytest.param("torch", marks=pytest.mark.torch),
+        ],
+    )
+    @pytest.mark.parametrize("op", ch_list)
+    @pytest.mark.parametrize("p", [0.0, 0.1, 1.0])
     @pytest.mark.parametrize("tr_args", [[100e-6, 100e-6, 20e-9], [100e-6, 120e-6, 20e-9]])
-    def test_kraus_matrices_sum_identity(self, ops, p, tr_args, tol):
+    def test_kraus_matrices_sum_identity(self, op, p, tr_args, interface, tol):
         """Test channels are trace-preserving"""
-        if ops.__name__ == "GeneralizedAmplitudeDamping":
-            op = [ops(p, p, wires=0)]
-        elif ops.__name__ == "ResetError":
-            op = [ops(p / 2, p / 3, wires=0)]
-        elif ops.__name__ == "PauliError":
-            op = [ops("X", p, wires=0), ops("XX", p, wires=[0, 1])]
-        elif ops.__name__ == "ThermalRelaxationError":
-            op = [ops(p, *tr_args, wires=0)]
+        p = qml.math.array(p, like=interface)
+        if op.__name__ == "GeneralizedAmplitudeDamping":
```

Comment:
Not a blocking request, but this may be a good opportunity to improve this test. Generally, if-blocks with more than three cases are "smelly". I almost vote we break this up into multiple tests in a class.

For one thing, `GeneralizedAmplitudeDamping` and `ResetError` are only getting tested when their two variables are matching. That is bound to let edge cases pass undetected.

We are also always looping over `tr_args`, but it's only used by one kind of operation.

Non-blocking, but always a good idea to leave code cleaner than you found it.

Reply:
Couldn't agree more :) I'll have a go.

Reply:
I reduced the case distinctions by a good bit. One could also go for multiple tests, but they would look completely identical up to the operation instantiation and the combinations of arguments. Therefore I went with two minor case distinctions but a single test for now.